components/playlist/playlist_data_source.cc

```diff
@@ -91,8 +132,35 @@ void PlaylistDataSource::StartDataRequest(
   GetDataFile(data_path, std::move(got_data_callback));
 }
 
+void PlaylistDataSource::RequestData(const GURL& url,
+                                     GotDataCallback got_data_callback) {
+  request_helper_->RequestRawData(
```

Comment:
Do I have to avoid making requests when `url` isn't `https` as per security checklist? cc. @brave/sec-team